### PR TITLE
[WIP] Add cover art api

### PR DESCRIPTION
**Pull request by shivam-kapila**
Add API to fetch and store cover art.

### CODE REVIEW

File: listenbrainz/db/cover_art.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            recording_msid: the MessyBrainz ID of the recording
+            image_url: the URL for the cover art
+    """
+
+    with db.engine.connect() as connection:
+        connection.execute(sqlalchemy.text("""
+            INSERT INTO cover_art (recording_msid, image_url)
+                 VALUES (:recording_msid, :image_url)
+            ON CONFLICT (recording_msid)
+          DO NOTHING
```
Comment (paramsingh) [inline listenbrainz/db/cover_art.py:18]: what happens if the url changes and we want to update it?

File: admin/sql/create_indexes.sql [hunk 2]
```diff
@@ -22,4 +22,6 @@ CREATE INDEX last_saved_ndx_follow_list ON follow_list (last_saved DESC);
 
 CREATE UNIQUE INDEX user_id_rec_msid_ndx_feedback ON recording_feedback (user_id, recording_msid);
 
+CREATE UNIQUE INDEX rec_msid_ndx_cover_art ON cover_art (recording_msid);
```
Comment (paramsingh) [inline admin/sql/create_indexes.sql:25]: I think this should be unique on msid, source not just msid

File: admin/sql/updates/2020-08-15-add-cover-art-table.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
+BEGIN;
+
+CREATE TABLE cover_art (
+    id                      SERIAL, -- PK
+    recording_msid          UUID NOT NULL,
+    image_url               VARCHAR NOT NULL
+);
+
+ALTER TABLE cover_art ADD CONSTRAINT cover_art_pkey PRIMARY KEY (id);
+
```
Comment (paramsingh) [inline admin/sql/updates/2020-08-15-add-cover-art-table.sql:10]: Don't see an index here

File: listenbrainz/db/cover_art.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            """), {
+                'recording_msid': recording_msid,
+                'image_url': image_url,
+            }
+        )
+
+
+def get_cover_art_url_for_recording(recording_msid: str):
```
Comment (paramsingh) [inline listenbrainz/db/cover_art.py:26]: If you're doing type hints, please add a return value for the function as well.

**Comment by mayhem** [comment]
Closing as per IRC discussion.

**Pull request closed without merging**